magenta-lib/src/main/scala/magenta/artifact/S3Location.scala

```diff
@@ -0,0 +1,89 @@
+package magenta.artifact
+
+import java.io.File
+
+import com.amazonaws.services.s3.AmazonS3
+import com.amazonaws.services.s3.model.AmazonS3Exception
+import com.gu.management.Loggable
+import magenta.{Build, DeployReporter}
+
+trait S3Location {
+  def bucket: String
+  def prefix: String
+  def prefixElements: List[String] = prefix.split("/").toList
+  def fileName:String = prefixElements.last
+  def extension:Option[String] = if (fileName.contains(".")) Some(fileName.split('.').last) else None
+  def string()(implicit client:AmazonS3):Option[String] = {
```

Comment:
Could rename this method to give more of a clue that it actually downloads data from S3.

Actually it doesn't seem to be used anywhere, so could be deleted?